python/semantic_kernel/connectors/memory/qdrant/qdrant_memory_store.py

```diff
@@ -0,0 +1,384 @@
+            )
+
+        self._qdrantclient = qdrant_client(host=hostip, port=port)
+        self._logger = logger or NullLogger()
+
+    async def create_collection_async(
+        self, collection_name: str, vector_size: int, distance: Optional[str] = "Cosine"
+    ) -> None:
+        """Creates a new collection if it does not exist.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to create.
+            vector_size {int} -- The size of the vector.
+            distance {Optional[str]} -- The distance metric to use. (default: {"Cosine"})
+        Returns:
+            None
+        """
+
+        self._qdrantclient.recreate_collection(
+            collection_name=collection_name,
+            vectors_config=VectorParams(size=vector_size, distance=Distance.COSINE),
+        )
+
+    async def get_collections_async(
+        self,
+    ) -> List[str]:
+        """Gets the list of collections.
+
+        Returns:
+            List[str] -- The list of collections.
+        """
+        return list(self._qdrantclient.get_collections())
+
+    async def delete_collection_async(self, collection_name: str) -> None:
+        """Deletes a collection.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to delete.
+
+        Returns:
+            None
+        """
+
+        self._qdrantclient.delete_collection(collection_name=collection_name)
+
+    async def does_collection_exist_async(self, collection_name: str) -> bool:
+        """Checks if a collection exists.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to check.
+
+        Returns:
+            bool -- True if the collection exists; otherwise, False.
+        """
+
+        collection_info = self._qdrantclient.get_collection(
+            collection_name=collection_name
+        )
+
+        if collection_info.status == CollectionStatus.GREEN:
+            return collection_name
+        else:
+            return ""
+
+    async def upsert_async(self, collection_name: str, record: MemoryRecord) -> str:
+        """Upserts a record.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to upsert the record into.
+            record {MemoryRecord} -- The record to upsert.
+
+        Returns:
+            str -- The unique database key of the record.
+        """
+        record._key = record._id
+
+        collection_info = self.does_collection_exist_async(
+            collection_name=collection_name
+        )
+
+        if not collection_info:
+            raise Exception(f"Collection '{collection_name}' does not exist")
+
+        upsert_info = self._qdrantclient.upsert(
+            collection_name=collection_name,
+            wait=True,
+            points=[
+                PointStruct(
+                    id=record._id, vector=record._embedding, payload=record._payload
```

Comment:
there's no need to use record._id here. It's independent of of record._key and record._key should be the pointID

Reply:
I don't see a resolution for this @tawalke -- thoughts?

Reply:
Resolved in https://github.com/microsoft/semantic-kernel/pull/919/commits/67f0898cddb7d290bb4b6511f7aaeeceaa63cd7a